packages/block-editor/src/store/actions.js

```diff
@@ -872,6 +873,30 @@ export const __unstableSplitSelection =
 			typeof selectionB.attributeKey === 'string'
 				? selectionB.attributeKey
 				: findRichTextAttributeKey( blockBType );
+		const blockAttributes = select.getBlockAttributes(
+			selectionA.clientId
+		);
+		const bindings = blockAttributes?.metadata?.bindings;
+
+		// If the attribute is bound, don't split the selection and insert a new block instead.
+		if ( bindings?.[ attributeKeyA ] ) {
+			// Show warning if user tries to insert a block into another block with bindings.
+			if ( blocks.length ) {
+				const { createWarningNotice } =
+					registry.dispatch( noticesStore );
+				createWarningNotice(
+					__(
+						"Blocks can't be inserted into other blocks with bindings"
```

Comment:
You can paste a paragraph inside a binded paragraph to update its content right?

I could do that on my testing, so in that case, we may change the warning to 
 `Blocks can't be insterted into different blocks with bindings`

Reply:
In my testing, it largely depend when you try to paste blocks. For example, try it with the Heading block, the Verse block, or caption for the Image block. It will get handled differently without any notices in the UI:


https://github.com/WordPress/gutenberg/assets/699132/249b096e-9d2f-4cff-864c-177e7014d443

In effect, my recommendation would be to take some action that fits best in the given scenario and skip the warning.

Reply:
Ideally, we probably want to modify the source value with whatever is supposed to be pasted. But that seemed a bit tricky to me at this point and I thought it could be handled in a follow-up PR. That's why I added the warning in the meantime.